plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -248,6 +248,9 @@ private static Optional<LastCheckpoint> findCheckpoint(TrinoFileSystem fileSyste
                 }
 
                 long version = checkpoint.get().version();
+                if (version == endVersion) {
```

Comment:
pre-existing: i noticed while reading the PR that we reuse the class `LastCheckpoint` which did correspond to what we used to do in Delta Lake while we used to be able to read only the latest checkpoint, but now this is not anymore the case.

Not for the current PR, rather a follow-up - come up with a better naming scheme for dealing with checkpoints.